### PR TITLE
[repo/AwsInstrumentations] Prepare to .NET9

### PR DESCRIPTION
**Pull request by Kielek**
## Changes

Follow up #2251

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* ~~[ ] Unit tests added/updated~~
* ~~[ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes~~
* ~~[ ] Changes in public API reviewed (if applicable)~~

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSMessagingUtils.cs [hunk 1]
```diff
@@ -89,33 +89,23 @@ internal static PropagationContext ExtractParentContext(SNSEvent.SNSMessage? mes
...
-                // Multiple values are stored as CSV (https://docs.aws.amazon.com/sns/latest/dg/sns-message-attributes.html).
-                return attribute.Value.Split(',');
-            default:
-                return null;
-        }
+        return !attributes.TryGetValue(attributeName, out var attribute)
+            ? null
+            : attribute?.Type switch
+            {
+                SnsAttributeTypeString when attribute.Value != null => (IEnumerable<string>?)[attribute.Value],
```
Comment (Oberon00) [inline src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSMessagingUtils.cs:103]: Is the IEnumerable cast really required here? The old `new[]` looked more readable (and shorter) to me in this instance...

File: src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSMessagingUtils.cs [hunk 2]
```diff
@@ -89,33 +89,23 @@ internal static PropagationContext ExtractParentContext(SNSEvent.SNSMessage? mes
...
+                SnsAttributeTypeStringArray when attribute.Value != null =>
+                    attribute.Value
+                        .Split(','), // Multiple values are stored as CSV (https://docs.aws.amazon.com/sns/latest/dg/sns-message-attributes.html).
```
Comment (Oberon00) [inline src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSMessagingUtils.cs:106]:
```suggestion
                SnsAttributeTypeStringArray when attribute.Value != null =>
                    // Multiple values are stored as CSV (https://docs.aws.amazon.com/sns/latest/dg/sns-message-attributes.html).
                    attribute.Value.Split(','), 
```

**Review by Oberon00 — APPROVED** [review]
Approving just for the Lambda part

File: src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSMessagingUtils.cs (hunk 1 above)
Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSMessagingUtils.cs]: Fixed in dcda60f6381b9cfea4ff88a5966dc377573f1987

**Review by ppittle — APPROVED** [review]

**Review by CodeBlanch — APPROVED** [review]
LGTM

File: src/OpenTelemetry.Instrumentation.AWS/Implementation/Utils.cs [hunk 3]
```diff
@@ -22,17 +22,12 @@ internal class Utils
...
-            return string.Empty;
-        }
-
-        if (originalString.EndsWith(suffix, StringComparison.Ordinal))
-        {
-            return originalString.Substring(0, originalString.Length - suffix.Length);
-        }
-
-        return originalString;
+        return string.IsNullOrEmpty(originalString)
```
Comment (muhammad-othman) [inline src/OpenTelemetry.Instrumentation.AWS/Implementation/Utils.cs:25]: nit: Do we have to make this change? The previous version was more readable to me.

**Pull request merged**